### PR TITLE
New MFEM version: 4.2

### PR DESCRIPTION
**Pull request by v-dobrev**
MFEM v4.2

### CODE REVIEW

**Review by tzanio — APPROVED** [review]
Thanks @v-dobrev !

**Comment by nghiakvnvsd** [comment]
Sorry to bother you. When I build MFEM-4.2 with spack, I don't see the variant, amgx. I don't know if you missed it?

File: var/spack/repos/builtin/packages/mfem/package.py [hunk 1]
```diff
@@ -45,10 +45,14 @@ class Mfem(Package):
     # other version.
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mfem/package.py:46]:
```suggestion
    version('master', branch='master')
```

**Review by acfisher — APPROVED** [review]

File: var/spack/repos/builtin/packages/mfem/package.py (hunk 1 above)
Comment (v-dobrev) [reply var/spack/repos/builtin/packages/mfem/package.py:46]: Changing `develop` to `master` will require changes in other packages (like `laghos`, `remhos`) and offline scripts that people may already have, so I'd prefer to keep it as is.

@tzanio, @acfisher, what do you think?

**Comment by v-dobrev** [comment]
> Sorry to bother you. When I build MFEM-4.2 with spack, I don't see the variant, amgx. I don't know if you missed it?

We plan to add the `amgx` variant in the near future but not in this PR.

**Review by balay — APPROVED** [review]

File: var/spack/repos/builtin/packages/mfem/package.py (hunk 1 above)
Comment (tzanio) [reply var/spack/repos/builtin/packages/mfem/package.py:46]: I'd prefer to keep `develop` as is too (it is also semantically correct).

We can add another version `master` if required.

**Pull request merged**